binary_wheel_builder/api/build.py

```diff
@@ -106,30 +108,43 @@ def build_wheel(wheel_meta: Wheel, dist_folder: Path, worker_count: int = 1) ->
     :return: Yields for each generated platform wheel
     """
     dist_folder.mkdir(exist_ok=True)
-    with concurrent.futures.ThreadPoolExecutor(max_workers=worker_count) as executor:
+  
+    worker_count = worker_count or os.cpu_count()
+    with concurrent.futures.ProcessPoolExecutor(max_workers=worker_count) as executor:
         futures = [
             executor.submit(
                 _build_wheel_for_platform,
                 dist_folder,
                 python_platform,
                 wheel_meta
             )
-            for python_platform in wheel_meta.platforms
-        ]
-        for future in concurrent.futures.as_completed(futures):
-            yield future.result()
-
-
+            for future in concurrent.futures.as_completed(futures):
+                if future.exception() is not None:
+                   logger.error(f"Sorry, a problem has occurred :(. Ensure all data is correct 
+                   or configured. Exception: {future.exception()}", exc_info=True)
+                else:
+                    try:
+                        yield future.result()
+                    except Exception as e:
+                       logger.error(f"Unexpected result has ocurred", exc_info=True)
```

Comment:
nit: Rephrase this to `Unexpected error`  which is more well known to users unfamiliar with python and, in general, more well established